06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,172 @@
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.globalIndex = {};
+  }
+
+  /**
+   * A method that converts string into an array
+   * @param  {string} word the string  from the "text" key in the JSON
+   * @return {Array} containing String.
+   */
+  static tokenize(word) {
+    const terms = word.replace(/[^\w\s]/gi, ' ')
+      .match(/\w+/g);
+    return terms;
+  }
+
+  /**
+   * A method that filters the array passed into it for unique words
+   * @param {Array} words Array of strings
+   * @returns {Array} Returns array of unique words
+   */
+  static uniqueWords(words) {
+    const unique = [...new Set(words)];
+    return unique;
+  }
+
+  /**
+   * A method that tokenizes an object and gets the unique terms in the object
+   * @param  {Object} file the uploaded JSON file object
+   * @return {Array}  of the unique terms in the file
+   */
+  static getText(file) {
+    let newText = '';
+    const space = ' ';
+    Object.keys(file)
+      .map((index) => {
+        newText += `${file[index].text} ${space}`;
+        return newText;
+      });
+    const terms = InvertedIndex.tokenize(newText);
+    const uniqueTerms = InvertedIndex
+      .uniqueWords(terms)
+      .map(x => x.toLowerCase());
+    return uniqueTerms;
+  }
+
+  /**
+   * A method to validate a JSON file.
+   * it check if the format of the .json file contains
+   * strictly keys of ["title","text"]
+   * @param {Object} parsedFile object
+   * @returns {Object} containing boolean and a String.
+   */
+  static validateFile(parsedFile) {
+    let isValid = {
+      valid: false,
+      message: 'This is an Invalid JSON File',
+    };
+    try {
+      Object.keys(parsedFile)
+        .forEach((key) => {
+          const uploadedFile = parsedFile[key];
+          const validFormat = ['title', 'text'];
+          const parsedFileFormat = Object.keys(uploadedFile);
+          const fileTextKey = Object.keys(parsedFileFormat)
+            .map(objKeys => parsedFileFormat[objKeys]);
+          if (validFormat.toString() === fileTextKey.toString()) {
+            isValid = {
+              valid: true,
+              message: 'This JSON Format is correct',
+            };
+          } else {
+            isValid = {
+              valid: false,
+              message: 'This JSON Format is Incorrect',
+            };
+            throw isValid;
+          }
+        });
+    } catch (error) {
+      if (error.valid === false) return error;
+      throw error;
+    }
+    return isValid;
+  }
+
+  /**
+   * A method that create the index of the selected-file
+   * @param  {Object} file the uploaded file object
+   * @param  {String} fileName    The file name
+   * @return {Object} containing the index of the selected-file
+   */
+  createIndex(file, fileName) {
+    const indexedFiles = {};
+    const uniqueTerms = InvertedIndex.getText(file);
+    uniqueTerms.forEach((uniqueKeys) => {
+      const arr = [];
+      file.forEach((jsonObjText) => {
```

Comment:
Thank you for pointing this out, I have rectified this issue, by using the correct variable naming convention.